### PR TITLE
Data and reviews

### PR DESCRIPTION
**Pull request by ArcasCZ**
- [x] Fix build
- [x] Create modul for importing / managing teacher and subject data
- [x] Port subject review function
- [x] Create teacher review function
- [x] CZ Translation
- [x] Cleanup
- [x] Move some things to core
- [x] Test all functions

### CODE REVIEW

**Comment by ArcasCZ** [comment]
Since it's kinda big and hard to test, I still expect some minor problems. But in this state I'm ready for review and after that for my final testing in service.

Also, some things in this module are using functions from https://github.com/pumpkin-py/pumpkin-py/pull/229

**Comment by ArcasCZ** [comment]
I've just made one change in module - there now can be more than one URL for subject (in case of mergin subjects with same abbreviation together)

**Comment by ArcasCZ** [comment]
Latest changes:

- If review was lastly edited before now and it's edited again, reset relevance. Users are voting for review and if that review is changed after a long period of time (e.g. if you failed that subject and you have to take it again), those user might not agree with this changes.
- Changed ordering of reviews - now primarily they are ordere by relevance and then by it's date.

File: po/cs.popie
Comment (xnomas) [inline po/cs.popie]: This doesn't feel like good English. 

`Your positive vote has been counted`...`noted`... but not `casted`. 

Same goes for the `negative vote`.

File: po/cs.popie
Comment (xnomas) [inline po/cs.popie]: `Can't remove teachers that don't teach this subject:`

File: po/cs.popie
Comment (xnomas) [inline po/cs.popie]: `A program with the abbreviation '{abbreviation}' and '{degree}' already exists.`

File: po/cs.popie
Comment (xnomas) [inline po/cs.popie]: `a JSON file`

File: po/sk.popie
Comment (xnomas) [inline po/sk.popie]: See above.

File: review/module.py
Comment (xnomas) [inline review/module.py]: Maybe consider adding a max length check? Same goes for Teacher reviews.

File: review/module.py
Comment (xnomas) [inline review/module.py]: The acl level check needs to change. Giving permissions to **all** members to override the reviews of other members shouldn't be allowed. 

I would change this to `ACLevel.SUBMOD`. 

same for the `review_subject_sudo_remove` command.

File: school/module.py
Comment (xnomas) [inline school/module.py]: The `N` key feels redundant... I don't see it in any of the popie strings or anywhere else really... do we need it?

File: school/module.py
Comment (xnomas) [inline school/module.py]: See above for the `N` key.

File: teacherchannel/database.py
Comment (xnomas) [inline teacherchannel/database.py]: I'd recommend learning some manners and not sending PRs that have diffs over 3000lines...  the maintainers have lives and stuff to do. Send smaller PRs and wait for Reviews. 

Cheers.

**Review by xnomas — CHANGES_REQUESTED** [review]

File: po/cs.popie
Comment (ArcasCZ) [reply po/cs.popie]: Fixed.

File: po/cs.popie
Comment (ArcasCZ) [reply po/cs.popie]: Fixed.

File: po/cs.popie
Comment (ArcasCZ) [reply po/cs.popie]: Fixed.

File: po/cs.popie
Comment (ArcasCZ) [reply po/cs.popie]: Fixed.

File: po/sk.popie
Comment (ArcasCZ) [reply po/sk.popie]: Fixed.

File: review/module.py
Comment (ArcasCZ) [reply review/module.py]: I think this is not necessary since Discord limits message length and I'm counting with long review text when preparing review embed list.

Also, this will be reworked since DC plans to add some forms and inputs as I've seen.

File: review/module.py
Comment (ArcasCZ) [reply review/module.py]: I see.

File: school/module.py
Comment (ArcasCZ) [reply school/module.py]: It's not because of the way we are getting data about subjects. Sometimes it's shown as N (Navazující), sometimes as M (Masters).

File: teacherchannel/database.py
Comment (ArcasCZ) [reply teacherchannel/database.py]: I'd like to see how would you cut the whole module into smaller PR's. It might be possible to make School and Review functions as two PR's, but since it's interconnected and it was constantly reworked during the development, it was not possible.

**Pull request closed without merging**
